projects/rero/ng-core/src/lib/record/record.service.ts

```diff
@@ -68,7 +69,14 @@ export class RecordService {
       httpParams = httpParams.append(key, preFilters[key]);
     }
 
-    return this.http.get<Record>(this.apiService.getEndpointByType(type, true) + '/', { params: httpParams })
+    let httpHeaders: HttpHeaders;
+    if (headers) {
+      httpHeaders = new HttpHeaders(headers);
```

Comment:
this can be done with an inline if statement.

Reply:
done, and I created a method to factorize this.